src/trio/_core/_parking_lot.py

```diff
@@ -234,6 +269,23 @@ def repark_all(self, new_lot: ParkingLot) -> None:
         """
         return self.repark(new_lot, count=len(self))
 
+    def break_lot(self, task: Task) -> None:
+        """Break this lot, causing all parked tasks to raise an error, and any
+        future tasks attempting to park (and unpark? repark?) to error. The error
```

Comment:
Should `ParkingLot.unpark`/`repark` raise `BrokenResourceError` if the lot is broken? Breaking the lot empties it, so `unpark`/`repark` becomes no-ops otherwise.

Reply:
I don't think so...

It's hard to think of a concrete case that this matters for; locks will not be able to reach them. However, it's possible to come up with cases I suppose?:
 - case for raising errors: data structure that needs to have at least one task parked at all times. Obviously having none would be a resource error.
 - case against: people putting parks in a try, then unparking the next task in a finally. Throwing would just make the exception worse.

I think the second case convinces me that it shouldn't raise but idk.

Reply:
yeah, I think treating these as noops makes sense, even though my initial inclination was to error out.